### ISSUE TITLE
add notion of a "primary" CompleteMoab for a druid

### ISSUE THREAD

**Issue by jmartin-sul** [sul-dlss/preservation_catalog#1528]
_**first thing to do:  pair w/ @jmartin-sul to discuss the two implementation options and pick one**_

For every druid in Preservation Catalog, there should be at least one `CompleteMoab` among our `MoabStorageRoot`s (exactly one at present, but adding proper support for more than one is the point of the epic that contains this ticket).  Further, there should be _exactly_ one `CompleteMoab` that is seen as the "primary".

The main purpose of this notion of "primary"-ness is to indicate which copy the Preservation Robots should update when an existing object is being versioned.

#### possible implementation approaches
_edit: went with a join table after some discussion, see #1577 for implementation, max one primary per druid is enforced by unique `preserved_object_id` col on the join table (in addition to its own primary key id)_

To whoever picks this up, I think some pairing and discussion about those implementation options is warranted, unless I'm missing something obvious that'd be easier than what I described there (which would be welcome).

#### rationale

In discussing how we'd implement multiple copies of a moab for a given druid, we settled on an approach where Preservation Robots would only update one copy (to keep down the required time and potential for error when robots are writing a large new version, e.g. a version that has a ton of video or something).  Since the main use case we're targeting is moving moabs from old storage to new storage, by way of making a copy and cutting over to the newer copy as the primary at some point, we expect that keeping the non-primary copy in sync will be the responsibility of some process outside of Preservation Catalog or Preservation Robots.  For example, an automated `rsync` of old storage root contents to new storage hardware, a manual copy by a person moving a moab to a storage root with more room for growth, or eventually, the ongoing `rsync` that backs up our primary storage brick's contents to our secondary storage brick.

#### next steps (for consideration in e.g. #1536)

We may also want to differentiate when audits discover unexpected inconsistencies.  E.g., if the "primary" moab is of a lower version than what's expected (lower than what we see on disk, lower than what `preserved_objects.current_version` indicates), our audits should alert loudly.  But if a non-primary copy is behind in version, we may want to log something and skip, e.g., the Honeybadger alert.

**Comment by jmartin-sul** [comment]
i'm marking this blocked by #366 and #1276.  i don't think this is impossible without first doing those, but i think this will be easier if we can make progress on those (also, i can imagine merge conflicts if all of those are being worked simultaneously).

if we can't make significant progress on those blocker issues in a couple days, we can unblock this and go for it.  i don't think it's absolutely required that we get those things done first, just very nice if possible.

**Comment by jmartin-sul** [comment]
closing in favor of #1536, #1583, and #1529
